processing/src/main/java/io/druid/query/aggregation/AggregatorFactory.java

```diff
@@ -89,27 +106,76 @@
    *
    * @return the finalized value that should be returned for the initial query
    */
-  public Object finalizeComputation(Object object);
+  public abstract Object finalizeComputation(Object object);
 
-  public String getName();
+  public abstract String getName();
 
-  public List<String> requiredFields();
+  public abstract List<String> requiredFields();
 
-  public byte[] getCacheKey();
+  public abstract byte[] getCacheKey();
 
-  public String getTypeName();
+  public abstract String getTypeName();
 
   /**
    * Returns the maximum size that this aggregator will require in bytes for intermediate storage of results.
    *
    * @return the maximum number of bytes that an aggregator of this type will require for intermediate result storage.
    */
-  public int getMaxIntermediateSize();
+  public abstract int getMaxIntermediateSize();
 
   /**
    * Returns the starting value for a corresponding aggregator. For example, 0 for sums, - Infinity for max, an empty mogrifier
    *
    * @return the starting value for a corresponding aggregator.
    */
-  public Object getAggregatorStartValue();
+  public abstract Object getAggregatorStartValue();
+
+  /**
+   * Merges the list of AggregatorFactory[] (presumable from metadata of some segments being merged) and
+   * returns merged AggregatorFactory[] (for the metadata for merged segment).
+   * Null is returned if it is not possible to do the merging for any of the following reason.
+   * - one of the element in input list is null i.e. aggregators for one the segments being merged is unknown
+   * - AggregatorFactory of same name can not be merged if they are not compatible
+   *
+   * @param aggregatorsList
+   *
+   * @return merged AggregatorFactory[] or Null if merging is not possible.
+   */
+  public static AggregatorFactory[] mergeAggregators(List<AggregatorFactory[]> aggregatorsList)
+  {
+    if (aggregatorsList == null || aggregatorsList.isEmpty()) {
+      return null;
+    }
+
+    Map<String, AggregatorFactory> mergedAggregators = new LinkedHashMap<>();
+
+    for (AggregatorFactory[] aggregators : aggregatorsList) {
+
+      if (aggregators != null) {
+        for (AggregatorFactory aggregator : aggregators) {
+          String name = aggregator.getName();
+          if (mergedAggregators.containsKey(name)) {
+            AggregatorFactory other = mergedAggregators.get(name);
+            try {
+              mergedAggregators.put(name, other.getMergingFactory(aggregator));
+            }
+            catch (AggregatorFactoryNotMergeableException ex) {
+              log.warn(ex, "failed to merge aggregator factories");
```

Comment:
oh, nevermind, it already does due to the automatic message in AggregatorFactoryNotMergeableException.